src/soap.service.ts

```diff
@@ -35,4 +42,24 @@ export class SoapService {
       return null;
     }
   }
+
+  private getAuthMethod(authOptions: AuthType): ISecurity {
+    const { username, password, type } = authOptions;
+
+    switch (type) {
+      case WSSECURITY_AUTH:
+        const WSSOptions = (authOptions as WSSecurityAuth).options;
+        return new WSSecurity(username, password, WSSOptions);
+      case SSL_PFX_AUTH:
+        const SSLPFXOptions = (authOptions as SSLPFXAuth).options;
+        const loginData = {
+          pfx: SSLPFXOptions.pfx,
+          passphrase: SSLPFXOptions.passphrase,
+          ...SSLPFXOptions.defaults,
+        };
+        return new ClientSSLSecurityPFX(loginData);
```

Comment:
Are you sure this is working? The `ClientSSLSecurityPFX` expects two or three args in the constructor and it's receiving only one. The typescript is not warning because the `loginData` is of `any` type and then it gets confused 🤷🏻.

I believe it would be better to do something like:
```typescript
return new ClientSSLSecurityPFX(
  SSLPFXOptions.pfx,
  SSLPFXOptions.passphrase,
  ...SSLPFXOptions.defaults
);
```